Parsers/CylancePROTECT/CylancePROTECT.txt

```diff
@@ -0,0 +1,144 @@
+        DeviceId = extract(@"Device Id: ([^,]+),",1,SyslogMessage),
+        PolicyName = extract(@"Policy Name: ([^,]+)(,|$)",1,SyslogMessage);
+let ThreatEvents = LogHeader
+| where EventType == "Threat"
+| extend SrcIpAddr = split(extract(@"IP Address: \(([^)]+)\),",1,SyslogMessage),","),
+        FileName = extract(@"File Name: ([^,]+),",1,SyslogMessage),
+        Path = extract(@"Path: ([^,]+)(,|$)",1,SyslogMessage),
+        DriveType = extract(@"Drive Type: ([^,]+),",1,SyslogMessage),
+        FileHashSha256 = extract(@"SHA256: ([^,]+),",1,SyslogMessage),
+        FileHashMd5 = extract(@"MD5: ([^,]+),",1,SyslogMessage),
+        Status = extract(@"Status: ([^,]+),",1,SyslogMessage),
+        CylanceScore = extract(@"Cylance Score: ([^,]+),",1,SyslogMessage),
+        FoundDate = extract(@"Found Date: ([^,]+),",1,SyslogMessage),
+        FileType = extract(@"File Type: ([^,]+),",1,SyslogMessage),
+        IsRunning = extract(@"Is Running: ([^,]+),",1,SyslogMessage),
+        AutoRun = extract(@"Auto Run: ([^,]+),",1,SyslogMessage),
+        DetectedBy = extract(@"Detected By: ([^,]+),",1,SyslogMessage),
+        ZoneNames = split(extract(@"Zone Names: \(([^)]+)\),",1,SyslogMessage),","),
+        IsMalware = extract(@"Is Malware: ([^,]+),",1,SyslogMessage),
+        IsUniqueToCylance = extract(@"Is Unique To Cylance: ([^,]+),",1,SyslogMessage),
+        ThreatClassification = extract(@"Threat Classification: ([^,]+),",1,SyslogMessage),
+        DeviceId = extract(@"Device Id: ([^,]+),",1,SyslogMessage),
+        PolicyName = extract(@"Policy Name: ([^,]+)(,|$)",1,SyslogMessage);
+let DeviceControlEvents = LogHeader
+| where EventType == "DeviceControl"
+| extend ExtDeviceType = extract(@"External Device Type: ([^,]+),",1,SyslogMessage),
+        ExtDeviceVendorId = extract(@"External Device Vendor ID: ([^,]+),",1,SyslogMessage),
+        ExtDeviceName = extract(@"External Device Name: ([^,]+),",1,SyslogMessage),
+        ExtDeviceProductId = extract(@"External Device Product ID: ([^,]+),",1,SyslogMessage),
+        ExtDeviceSerialNumber = extract(@"External Device Serial Number: ([^,]+),",1,SyslogMessage),
+        ZoneNames = split(extract(@"Zone Names: \(([^)]+)\)(,|$)",1,SyslogMessage),","),
+        DeviceId = extract(@"Device Id: ([^,]+),",1,SyslogMessage),
+        PolicyName = extract(@"Policy Name: ([^,]+)(,|$)",1,SyslogMessage);
+let AuditLog = LogHeader
+| where EventType == "AuditLog"
+| extend Provider = extract(@"Provider: ([^,]+),",1,SyslogMessage),
+        PolicyChanged = extract(@"Policy Changed: ([^,]+),",1,SyslogMessage),
+        ZoneNames = split(extract(@"Zone: ([^,]+)(,|$)",1,SyslogMessage),","),
+        UserName = extract(@"User: ([^,]+)(,|$)",1,SyslogMessage),
+        DeviceName = extract(@"Devices?:\s([\w\d\-\_]+)\,",1,SyslogMessage),
+        SrcIpAddr = split(extract(@"Source IP: ([\w\.\:]+)\,",1,SyslogMessage),",");
+let AllOtherLogs = LogHeader
```

Comment:
May want to consider populating the EventType field here with something like "OtherType' if it is empty.  It may help query efficiency later.

Reply:
@shainw added Events that do not have a Event Type listed as "Other. Also added the capture of other Event Types that may not be defined